### DISCUSSION TITLE
Is "Height" channel is always necessary?

### DISCUSSION THREAD

**Discussion by Wang-stables** [AFM-SPM/TopoStats#563] [Q&A] [answered]
It seems that program need the .spm file have height channel on it, otherwise it won't work.

**Comment by ns-rse** [comment, marked as answer]
Hi @Wang-stables 

Thanks for checking out TopoStats and getting in touch.

SPM files can have one of the following fields, you would set this in the `channel` field of a custom configuration file (see [here](https://afm-spm.github.io/TopoStats/main/configuration.html#fields)).

```
loading:
  channel: Height Sensor
```

You can generate configuration files if you [install from GitHub](https://afm-spm.github.io/TopoStats/main/installation.html#cloning-from-github) under a virtual environment and use the [`--create-config-file`](https://afm-spm.github.io/TopoStats/main/configuration.html#generating-a-configuration) flag to `run_topostats`.

We are due to release `v2.1.0` to PyPI in the coming days which will include all of the recent changes.

Let me know how you get on,

Neil

**Reply by Wang-stables** [reply]
I think I can understand what you are talking about, unfortunately I'm not familier to .yaml format yet, so I can't give you the further usage experience right now, I will update the information on this page when I clearly and successfully control this part of programming.
Thanks for your very fast reply and your team's hard work on this, have a nice day.

**Reply by neumagen** [reply]
![Screen Shot 2023-07-10 at 2 10 34 PM](https://github.com/AFM-SPM/TopoStats/assets/139154686/d9b7fc57-698d-43d2-b297-b96a33184748)

What are the different channel names available to use? I tried ZSensor, zsensor, Height Sensor, PeakForceError, [ZSensor, PeakForceError], and ['ZSensor', 'PeakForceError']. What am I missing?

**Reply by ns-rse** [reply]
Hi @neumagen 

The error message here is saying that the requested channel `Height` is not present in the `.spm` channel, but that the found channels are `ZSensor` and `PeakForceError` (these are read from the file itself).

Its surprising that using either of these in your configuration didn't work.

Could you share the YAML file with `ZSensor` in and the output that it gives too please?

Ideally if you could copy and paste the file and output it makes it easier to search the text and find things, particularly for the YAML file. You can make it look like a code by putting it in a code block as shown below.

````
```
This is a code block.
```
````

Thanks @ns-rse

**Reply by neumagen** [reply]
**EDIT** Moved to to a separate [thread](https://github.com/AFM-SPM/TopoStats/discussions/619).

**Reply by derollins** [reply]
For reasons I don't understand sometimes although it says Zsensor in the read-out you need to put `channel:  Height Sensor` in the  config file rather than zsensor.

**Reply by ns-rse** [reply]
@neumagen I think I may have worked out what the problem was, see the [separate thread](https://github.com/AFM-SPM/TopoStats/discussions/619).
